common/authen.js

```diff
@@ -404,7 +471,20 @@
                     // and resolve the promise to notify ermrestjs that the user has logged in
                     // and it can continue firing other queued calls
                     Session.getSession("401").then(function(_session) {
-                        defer.resolve();
+                        var prevSession = Session.getPrevSessionValue();
+                        // TODO: limiting this to entry may not be the right condition?
+                        // Not sure if this will trigger on page load, if not I think it's safe to run this in all contexts
+                        // recordset  - read should throw a 409 if there's a permission issue
+                        // record     - same issue with read above
+                        //            - p&b add throws a conflict error in most cases, so won't get sent here either (RBK)
+                        // recordedit - add/update return here in some cases, and in other cases will return a 409 and ignore this case
+                        var differentUser = (ConfigUtils.getContextJSON().appContext.indexOf("entry") != -1 && prevSession && !Session.isSameSessionAsPrevious())
```

Comment:
I don't think you need to check for `entry` context. This error applies to all the 401 error cases (regardless of the app) and as you mentioned even the recordedit case that we were trying to solve doesn't go through this and actually throws a 409.